app/lib/service/email/models.dart

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:clock/clock.dart';
+
+import '../../shared/datastore.dart' as db;
+import '../../shared/utils.dart';
+
+/// A record of an outgoing email to a single recipient.
+@db.Kind(name: 'OutgoingEmail', idType: db.IdType.String)
+class OutgoingEmail extends db.Model {
+  String get uuid => id as String;
+
+  @db.DateTimeProperty(required: true)
+  DateTime? created;
+
+  /// The total number of attempts.
+  @db.IntProperty(required: true, indexed: false)
+  int attempts = 0;
+
+  /// The timestamp of the last attempt.
+  @db.DateTimeProperty()
+  DateTime? lastAttempted;
```

Comment:
Why not `pendingAt`, which you can set to year 3000 when the message is dead and shouldn't be retried again.